src/category_theory/products/basic.lean

```diff
@@ -51,45 +51,30 @@ variables (C : Type u₁) [𝒞 : category.{v₁} C] (D : Type u₂) [𝒟 : cat
 include 𝒞 𝒟
 
 /-- `inl C Z` is the functor `X ↦ (X, Z)`. -/
-def inl (Z : D) : C ⥤ C × D :=
+@[simps obj map] def inl (Z : D) : C ⥤ C × D :=
```

Comment:
```suggestion
-- Here and below we specify explicitly the projections to generate `@[simp]` lemmas for, 
-- as the default behaviour of `@[simps]` will generate projections all the way down to components of pairs.
@[simps obj map] def inl (Z : D) : C ⥤ C × D :=
```

Reply:
Do you think we should explain the meaning of an attribute when we use it? I would much rather just have this explanation in the documentation (like `tactics.md`)

Reply:
Actually, I take that back. Your comment is pretty specific to these uses.